airflow/models/taskinstance.py

```diff
@@ -2553,21 +2554,30 @@ def _run_raw_task(
 
             return None
 
-    def _register_dataset_changes(self, *, session: Session) -> None:
+    def _register_dataset_changes(self, result: Any, *, session: Session) -> None:
         if TYPE_CHECKING:
             assert self.task
 
         for obj in self.task.outlets or []:
             self.log.debug("outlet obj %s", obj)
             # Lineage can have other types of objects besides datasets
             if isinstance(obj, Dataset):
+                if obj.extra:
+                    extra = obj.extra
```

Comment:
I do not think this is right. It’s made quite clear in previous issues that `Dataset.extras` and `DatasetEvent.extras` are different things and should be kept separate.

Reply:
Sorry, then it seems I mis-understood your comments in the previous PR7discussions about this. Thanks for clearly documenting the differences between `Dataset` and `DatasetEvent`(both) `extra` details in #38481. This opened my eyes and now I understand your push-back.
Now it is "clear" to me. Intent was _not_ to overwrite or mangle "static" information for events.